### PR TITLE
feat: Implement presign support

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR implement the following feature:

```sql
MySQL [(none)]> PRESIGN @my_stage/books.csv
+--------+---------+---------------------------------------------------------------------------------+
| method | headers | url                                                                             |
+--------+---------+---------------------------------------------------------------------------------+
| GET    | {}      | https://example.s3.amazonaws.com/books.csv?X-Amz-Algorithm=AWS4-HMAC-SHA256&... |
+--------+---------+---------------------------------------------------------------------------------+
```

The PR also adds stateful tests that:

- presign_download: Generate a URL and use curl to download an object
- presign_upload: Generate a URL and use curl to upload an object

Close https://github.com/datafuselabs/databend/issues/6215

### CODE REVIEW

**Review by junnplus — APPROVED** [review]

**Comment by ariesdevil** [comment]
LGTM, it would be nice to have the documentation :D

**Comment by Xuanwo** [comment]
> LGTM, it would be nice to have the documentation :D

I will add docs in the next PR~

File: tests/suites/1_stateful/03_presign/03_0000_presign_download.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "$CURDIR"/../../../shell_env.sh
+
+echo "drop stage if exists presign_stage" | $MYSQL_CLIENT_CONNECT
+
+aws --endpoint-url http://127.0.0.1:9900/ s3 cp s3://testbucket/admin/data/ontime_200.csv s3://testbucket/admin/stage/presign_stage/ontime_200.csv >/dev/null 2>&1
+
+echo "CREATE STAGE presign_stage;" | $MYSQL_CLIENT_CONNECT
```
Comment (BohuTANG) [inline tests/suites/1_stateful/03_presign/03_0000_presign_download.sh:10]: What will happen if we create an external stage, and get it pregisn?

Comment (Xuanwo) [reply tests/suites/1_stateful/03_presign/03_0000_presign_download.sh:10]: presign will generate a URL with the external stage's credential.

**Comment by BohuTANG** [comment]
Databend supports file function: `get_presigned` as [snow](https://docs.snowflake.com/en/sql-reference/functions/get_presigned_url.html), need documention cc @soyeric128

**Pull request merged**